Lib/test/test_capi/test_opt.py

```diff
@@ -1925,6 +1925,51 @@ def testfunc(n):
         self.assertNotIn("_GUARD_NOS_INT", uops)
         self.assertNotIn("_GUARD_TOS_INT", uops)
 
+    def test_get_len_with_const_tuple(self):
+        def testfunc(n):
+            x = ""
+            for _ in range(n):
+                match (1, 2, 3, 4):
+                    case [_, _, _, _]:
+                        x += "1"
+            return x
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(len(res), TIER2_THRESHOLD)
+        uops = get_opnames(ex)
+        self.assertNotIn("_GUARD_NOS_INT", uops)
+        self.assertIn("_GET_LEN", uops)
+        self.assertIn("_POP_TOP_LOAD_CONST_INLINE_BORROW", uops)
+
+    def test_get_len_with_non_const_tuple(self):
+        def testfunc(n):
+            x = ""
+            for _ in range(n):
+                match (object(), object()):
+                    case [_, _]:
+                        x += "1"
+            return x
+        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
+        self.assertEqual(len(res), TIER2_THRESHOLD)
```

Comment:
I don't love adding the strings like this. Can we just use floats? (In the other tests too):

```suggestion
            x = 0.0
            for _ in range(n):
                match (object(), object()):
                    case [_, _]:
                        x += 1.0
            return x
        res, ex = self._run_with_optimizer(testfunc, TIER2_THRESHOLD)
        self.assertEqual(res, TIER2_THRESHOLD)
```